### ISSUE TITLE
[java] UnusedImports false positive for Javadoc link with array type

### ISSUE THREAD

**Issue by davidburstromspotify** [pmd/pmd#1720]
**Affects PMD Version:** 
Confirmed on 6.12.0 and 6.8.0

**Rule:**
UnusedImports

**Description:**

In case a JavaDoc link refers to an array typed imported type, the import is flagged as redundant.

**Code Sample demonstrating the issue:**
`FileAttribute` will be flagged as redundant.
```
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.FileAttribute; 
/**
 * See {@link Files#createDirectory(Path, FileAttribute[])}
 */
public class Test {}
```

**Running PMD through:**
Gradle/CLI

**Comment by jsotuyod** [comment]
@davidburstromspotify thanks for the report! Effectively this is a false positive.

**Comment by itaigilo** [comment]
Fixed this at #1746.
